webui/src/pages/repositories/repository/objects.jsx

```diff
@@ -3,22 +3,230 @@ import React, {useRef, useState} from "react";
 import {useAPI, useAPIWithPagination} from "../../../lib/hooks/api";
 import {RefContextProvider, useRefs} from "../../../lib/hooks/repo";
 import {useRouter} from "../../../lib/hooks/router";
 import {RefTypeBranch} from "../../../constants";
+import Alert from "react-bootstrap/Alert";
+import {Link} from "react-router-dom";
+import {LinearProgress} from '@mui/material';
 
+const ImportProgress = ({ numObjects}) => {
+    return (
+        <Row>
+            <Col>
+                <div style={{display: 'flex', justifyContent: 'center'}}>
+                    <p>Imported <strong><div style={{ color: 'green', display: 'inline'}}> {numObjects} </div></strong> objects so far...</p>
+                </div>
+                <div>
+                    <LinearProgress color="success" />
+                </div>
+                <div>
+                    <small><abbr><br/>Please leave this tab open while import is in progress...</abbr></small>
+                </div>
+            </Col>
+        </Row>
+    );
+}
+
+const ImportDone = ({ numObjects, importBranch, currBranch}) => {
+    return (
+        <Row>
+            <Col>
+                <div className={"mt-10 mb-2 mr-2 row mt-4"} style={{color: 'green', display: 'flex', justifyContent: 'center', fontSize: '160%'}}>
+                    <p><strong>Success!</strong></p>
+                </div>
+                <div style={{display: 'flex', justifyContent: 'center'}}>
+                    <p><strong><div style={{ color: 'green', display: 'inline'}}> {numObjects} </div></strong> objects imported and committed into branch {importBranch}.</p>
+                </div>
+                <div style={{display: 'flex', justifyContent: 'center'}}>
+                    <p> Use the&nbsp;<Link to={`compare?ref=${ currBranch }&compare=${ importBranch }`} variant = "success" >Compare tab</Link>&nbsp;to view the changes and merge them to {currBranch}.</p>
+                </div>
+                <Alert variant="warning" className={"ml-2 mr-2 row mt-4"}>
+                    Merging will override the state of {currBranch} with the imported objects.
+                </Alert>
+
+            </Col>
+        </Row>
+    );
+}
 
-const UploadButton = ({ config, repo, reference, path, onDone, variant = "success", onClick, onHide, show = false}) => {
+const ImportButton = ({ config, repo, reference, path, onDone, onClick, variant = "success", onHide, show = false, enabled= false}) => {
+    const initialState = {
+        inProgress: false,
+        error: null,
+        done: false,
+        numObj: 0,
+        isStorageNamespaceValid: null
+    }
+    const [importState, setImportState] = useState(initialState)
+
+    const sourceRef = useRef(null);
+    const destRef = useRef(null);
+    const commitMsgRef = useRef(null);
+    const storageType = config.blockstore_type
+    const DEFAULT_BLOCKSTORE_VALIDITY_REGEX = new RegExp(`^s3://`);
```

Comment:
Why? We support Azure and GCP too..

Reply:
@eden-ohana @itaiad200 did we test this feature on Azure and GCS?

Reply:
yes